Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIScriptingCocoa.mm

```diff
@@ -78,31 +156,50 @@
         script = mergeDictionariesAndSetValues(script, @{ key: func.toString });
     }
 
-    // FIXME: <https://webkit.org/b/259954> May need to remove unsupported keys.
-    // FIXME: Handle script injection in the UI Process.
+    WebExtensionScriptInjectionParameters parameters;
+    parseTargetInjectionOptions(script[targetKey], parameters, outExceptionString);
+    parseScriptInjectionOptions(script, parameters);
+
+    WebProcess::singleton().sendWithAsyncReply(Messages::WebExtensionContext::ScriptingExecuteScript(WTFMove(parameters)), [protectedThis = Ref { *this }, callback = WTFMove(callback)](std::optional<Vector<WebExtensionScriptInjectionResultParameters>> results) {
+        if (!results) {
+            callback->call();
+            return;
+        }
+
+        callback->call(toAPI(results.value()));
+    }, extensionContext().identifier().toUInt64());
 }
 
-void WebExtensionAPIScripting::insertCSS(NSDictionary *cssInfo, Ref<WebExtensionCallbackHandler>&& callback, NSString **outExceptionString)
+void WebExtensionAPIScripting::insertCSS(WebPage* page, NSDictionary *cssInfo, Ref<WebExtensionCallbackHandler>&& callback, NSString **outExceptionString)
 {
```

Comment:
You pass `WebPage` in as a pointer, but never null-check it. If it's required that it is always non-null, please pass a reference. If it might be nullptr, please check for that.

Reply:
Alternatively, consider just passing the WebPageProxy identifier, which seem to be the only thing you need from it.

Reply:
This is called by the code generated code. It is pretty guaranteed to not be null since this comes from the JS world owned by the page, but I'll fix the code generator to pass references here instead. We do need the page sometimes, so it can't just be identifier.

Reply:
We actually don't need the WebPage in this case, so I'll remove.